lib/ansible/module_utils/acme.py

```diff
@@ -426,6 +426,15 @@ def _sign_request_cryptography(module, payload64, protected64, key_data):
         "signature": nopad_b64(signature),
     }
 
+def _assert_fetch_url_success(response, info, allow_redirect=False, allow_client_error=True, allow_server_error=True):
+    if info['status'] < 0:
+        raise ModuleFailException(msg="Failure downloading %s, %s" % (url, to_native(e)))
+
+    if (info['status'] >= 300 and info['status'] < 400 and not allow_redirect) or \
```

Comment:
You can simplify this line and the next ones as follows:
```suggestion
    if (300 <= info['status'] < 400 and not allow_redirect) or \
```
In Python (as opposed to C/C++), this yields the correct behavior.

Reply:
Didn't know about this syntax! Updated in commit https://github.com/ansible/ansible/pull/63140/commits/b1b619062ad495aa18ffeeb2c01056b06e7c7fac